mantidimaging/gui/windows/stack_properties_dialog/presenter.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (C) 2021 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING
+
+from mantidimaging.gui.mvp_base import BasePresenter
+
+if TYPE_CHECKING:
+    from mantidimaging.gui.windows.stack_properties_dialog.view import StackPropertiesDialog
+
+
+class StackPropertiesPresenter(BasePresenter):
+
+    def __init__(self, view: StackPropertiesDialog):
+        super().__init__(view)
+
+    def set_stack_from_data_type(self, origin_data_type) -> None:
+        if origin_data_type == "Sample":
+            self.view.stack = self.view.origin_dataset.sample
+        elif origin_data_type == "Flat Before":
+            self.view.stack = self.view.origin_dataset.flat_before
+        elif origin_data_type == "Flat After":
+            self.view.stack = self.view.origin_dataset.flat_after
+        elif origin_data_type == "Dark Before":
+            self.view.stack = self.view.origin_dataset.dark_before
+        elif origin_data_type == "Dark After":
+            self.view.stack = self.view.origin_dataset.dark_after
+
+    def set_stack_data(self):
+        if self.view.stack is not None:
+            if self.view.stack.filenames is not None:
+                self.view.directory = self.view.stack.filenames[0].replace(self.view.stack.filenames[0].split("\\")[-1],
+                                                                           '')
+            self.view.stack_shape = self.view.stack.data.shape
+        self.view.stack_size_MB = self.get_stack_size_MB()
+
+    def get_stack_size_MB(self):
+        return sum(
+            os.path.getsize(self.view.directory + f)
+            for f in os.listdir(self.view.directory) if os.path.isfile(self.view.directory + f)) / 1024 / 1024
```

Comment:
I get
```
  File "/home/sam/git/mantidimaging/mantidimaging/gui/windows/stack_properties_dialog/presenter.py", line 42, in get_stack_size_MB
    for f in os.listdir(self.view.directory) if os.path.isfile(self.view.directory + f)) / 1024 / 1024
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: ''
```

In general it would be better to do all the path handing with pathlib, e.g. `path.parent()` instead using string functions.

But here it would be better to use the memory size of the array rather than the file sizes. There are functions in `size_calculator.py` to do it.